Cannot find type definition file for 'chai' and 'typed-graphql'

See #823. The "fix" in that issue is not a real fix so that's why I'm opening a new issue. You need to add those packages as dependencies and not just devDependencies.

Same issue for me, Typescript 2.1.x

```
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/react-apollo/graphql.d.ts
(4,46): error TS2307: Cannot find module 'graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/react-apollo/graphql.d.ts
(1,1): error TS2688: Cannot find type definition file for 'chai'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/react-apollo/graphql.d.ts
(2,1): error TS2688: Cannot find type definition file for 'typed-graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/react-apollo/index.d.ts
(3,21): error TS2307: Cannot find module 'recompose/compose'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/fragments.d.ts
(2,46): error TS2307: Cannot find module 'graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/fragments.d.ts
(1,1): error TS2688: Cannot find type definition file for 'typed-graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/data/writeToStore.d.ts
(4,40): error TS2307: Cannot find module 'graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/data/writeToStore.d.ts
(1,1): error TS2688: Cannot find type definition file for 'chai'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/data/writeToStore.d.ts
(2,1): error TS2688: Cannot find type definition file for 'typed-graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/errors/ApolloError.d.ts
(3,30): error TS2307: Cannot find module 'graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/errors/ApolloError.d.ts
(2,1): error TS2688: Cannot find type definition file for 'typed-graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/ApolloClient.d.ts
(1,1): error TS2688: Cannot find type definition file for 'chai'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/core/QueryManager.d.ts
(8,26): error TS2307: Cannot find module 'graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/core/QueryManager.d.ts
(1,1): error TS2688: Cannot find type definition file for 'typed-graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/core/QueryManager.d.ts
(3,1): error TS2688: Cannot find type definition file for 'chai'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/core/watchQueryOptions.d.ts
(4,46): error TS2307: Cannot find module 'graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/core/watchQueryOptions.d.ts
(1,1): error TS2688: Cannot find type definition file for 'typed-graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/core/watchQueryOptions.d.ts
(2,1): error TS2688: Cannot find type definition file for 'chai'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/core/ObservableQuery.d.ts
(1,1): error TS2688: Cannot find type definition file for 'chai'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/data/readFromStore.d.ts
(3,26): error TS2307: Cannot find module 'graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/data/readFromStore.d.ts
(1,1): error TS2688: Cannot find type definition file for 'typed-graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/data/readFromStore.d.ts
(2,1): error TS2688: Cannot find type definition file for 'chai'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/data/extensions.d.ts
(1,1): error TS2688: Cannot find type definition file for 'chai'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/mutations/store.d.ts
(5,30): error TS2307: Cannot find module 'graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/mutations/store.d.ts
(1,1): error TS2688: Cannot find type definition file for 'chai'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/mutations/store.d.ts
(3,1): error TS2688: Cannot find type definition file for 'typed-graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/queries/getFromAST.d.ts
(2,67): error TS2307: Cannot find module 'graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/queries/getFromAST.d.ts
(1,1): error TS2688: Cannot find type definition file for 'typed-graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/data/scopeQuery.d.ts
(3,30): error TS2307: Cannot find module 'graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/data/scopeQuery.d.ts
(1,1): error TS2688: Cannot find type definition file for 'typed-graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/data/mutationResults.d.ts
(4,41): error TS2307: Cannot find module 'graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/data/mutationResults.d.ts
(1,1): error TS2688: Cannot find type definition file for 'typed-graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/data/mutationResults.d.ts
(2,1): error TS2688: Cannot find type definition file for 'chai'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/actions.d.ts
(4,41): error TS2307: Cannot find module 'graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/actions.d.ts
(1,1): error TS2688: Cannot find type definition file for 'typed-graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/actions.d.ts
(3,1): error TS2688: Cannot find type definition file for 'chai'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/queries/store.d.ts
(5,44): error TS2307: Cannot find module 'graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/queries/store.d.ts
(1,1): error TS2688: Cannot find type definition file for 'chai'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/queries/store.d.ts
(3,1): error TS2688: Cannot find type definition file for 'typed-graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/data/storeUtils.d.ts
(3,65): error TS2307: Cannot find module 'graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/data/storeUtils.d.ts
(1,1): error TS2688: Cannot find type definition file for 'typed-graphql'.errors @ main.js:672sock.onmessage @ main.js:2221EventTarget.dispatchEvent @ main.js:3682(anonymous function) @ main.js:6775SockJS._transportMessage @ main.js:6773EventEmitter.emit @ main.js:3618WebSocketTransport.ws.onmessage @ main.js:2307
main.js:672 /Users/Danijel/projects/am/am-web/node_modules/apollo-client/data/storeUtils.d.ts
(2,1): error TS2688: Cannot find type definition file for 'chai'.
```

--- a/package.json
+++ b/package.json
@@ -63,7 +63,6 @@
     "whatwg-fetch": "^2.0.0"
   },
   "devDependencies": {
-    "@types/chai": "^3.4.32",
     "@types/chai-as-promised": "0.0.28",
     "@types/mocha": "^2.2.31",
     "browserify": "^13.0.0",
@@ -87,17 +86,18 @@
     "sinon": "^1.17.4",
     "source-map-support": "^0.4.0",
     "tslint": "3.15.1",
-    "typed-graphql": "1.0.2",
     "typescript": "2.0.10",
     "uglify-js": "^2.6.2"
   },
   "optionalDependencies": {
     "@types/async": "^2.0.31",
+    "@types/chai": "^3.4.32",
     "@types/isomorphic-fetch": "0.0.30",
     "@types/lodash": "^4.14.34",
     "@types/node": "^6.0.38",
     "@types/promises-a-plus": "0.0.26",
     "@types/redux": "^3.5.29",
-    "@types/sinon": "^1.16.29"
+    "@types/sinon": "^1.16.29",
+    "typed-graphql": "1.0.2"
   }
 }

